### PR TITLE
History - Employment updates

### PR DESCRIPTION
**Pull request by floresj**
Updated the use of the `Reprimand` component so that it persisted to the DB. This was throwing off the validator logic. I also updated the date comparison logic to build the dates from the stored values. This was causing  the Reprimand component to not render because the function that toggled visibility was not computing the date logic properly.

### CODE REVIEW

File: src/components/Section/History/Employment/EmploymentItem.jsx [hunk 1]
```diff
@@ -199,9 +200,9 @@ export default class EmploymentItem extends ValidationElement {
   showLeaving () {
     const activity = (this.props.EmploymentActivity || {}).value
     const sevenYearsAgo = daysAgo(today, 365 * 7)
-    const from = (this.props.Dates || {}).from
-    const to = (this.props.Dates || {}).to
-    return (from && from.date >= sevenYearsAgo) || (to && to.date >= sevenYearsAgo) &&
+    const from = buildDate((this.props.Dates || {}).from)
```
Comment (bmallred) [inline src/components/Section/History/Employment/EmploymentItem.jsx]: :+1: for reuse!

**Comment by bmallred** [comment]
Looks like errors within the reference portion are not being taken in to consideration

image

**Comment by floresj** [comment]
Ahhhhh that's a good find. This may be a site wide thing. I realized that the nav strictly uses the validator logic to determine when to render the green dot. If there's an actual error, I don't think it'll get captured.  For instance, I filled out all the fields for the applicant name but put in an error and it still shows up green. While the content is invalid, from the validator perspective, it's still returning true because all fields are populated.

selection_731

Now, if I clear a value, the validator will return false which then just clears the green dot but doesn't capture that there's an actual error.

selection_732

We're probably going to have to incorporate the errors in the error store to get those red dots to show up.

**Comment by bmallred** [comment]
That's the thing... they should be in the error store. There was a similar issue for reference emails that were wrapped in `NotApplicable`. To resolve that one had to pass in the `onError` property to `NotApplicable` to get it to show up.

**Comment by bmallred** [comment]
**Note to self:** Going to finish this one up in the morning.

**Comment by floresj** [comment]
:+1: thanks for the updates.

**Pull request merged**